### PR TITLE
Fixes for cognitive services; Fixed issue #6054; Private virtual methods should be included in ApiView

### LINKED ISSUE #6054: ApiView for C++: support [[deprecated]] attribute
ApiView needs to support `[[deprecated]]` attribute.
This came as a comment in this Azure Core review: https://apiview.dev/Assemblies/Review/e70fdd6e25954ba8b474f7e7b0fc022b

### PR DESCRIPTION
**Pull request by LarryOsterman**
Fixes #6054

### CODE REVIEW

File: tools/apiview/parsers/cpp-api-parser/ParseTests/TestCases/ExpressionTests.cpp [hunk 1]
```diff
@@ -14,10 +14,12 @@ struct HelperClass
 class TestClass {
 public:
   std::string PublicField1;
+  [[deprecated]]
```
Comment (RickWinter) [inline tools/apiview/parsers/cpp-api-parser/ParseTests/TestCases/ExpressionTests.cpp:17]: We own this tool completely.  Can we simply remove?

File: tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp [hunk 2]
```diff
@@ -2295,12 +2581,26 @@ AstClassLike::AstClassLike(
     m_isAnonymousNamedStruct = true;
     m_anonymousNamedStructName = cast<FieldDecl>(decl->getNextDeclInContext())->getNameAsString();
   }
+  if (decl->getNameAsString() == "Md5Hash")
+  {
+    decl->dump(llvm::outs());
+  }
```
Comment (RickWinter) [inline tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:2587]: Why do we only care about Md5Hash?

Comment (RickWinter) [reply tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/AstNode.cpp:2587]: Add comment

File: tools/apiview/parsers/cpp-api-parser/ParseTests/TestCases/ExpressionTests.cpp (hunk 1 above)
Comment (LarryOsterman) [reply tools/apiview/parsers/cpp-api-parser/ParseTests/TestCases/ExpressionTests.cpp:17]: This is from the test case for the [[deprecated]] feature logic.

**Comment by LarryOsterman** [comment]
Ping :).

**Review by RickWinter — APPROVED** [review]

**Pull request merged**